WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift

```diff
@@ -623,6 +623,17 @@ private extension OrderDetailsViewController {
         }
     }
 
+    private func retryCollectPayment() {
+        viewModel.cancelPayment { [weak self] result in
+            switch result {
+            case .failure(let error):
+                self?.paymentAlerts.nonRetryableError(from: self, error: error)
```

Comment:
This is were the edge case mention in the PR description is: if there is a failure the fist time we attempt to cancel the previous payment attempt, we always transition to the non returnable alert.

If the first attempt to collect the payment failed because there is not a valid internet connection, the attempt to cancel the previous payment fails because we never got to even create a payment intent. 

If the internet connection comes back in between the first error and the time we hit retry, we would not be aware of it at this point, so we can not swallow the second error